src/runner.ts

```diff
@@ -0,0 +1,578 @@
+              `   ${git.serializeGitHubRepoRef(repo[1].githubRepoRef)}`);
+        }
+      }
+    }
+  }
+
+  /**
+   * Given a collection of GitHubRepoRefs, replace any that represent wildcard
+   * values with the literal values after comparing against names of repos on
+   * GitHub.  So a repo ref like `Polymer/*` return everything owned by
+   * Polymer where `PolymerElements/iron-*` would be all repos that start with
+   * `iron-` owned by `PolymerElements` org.
+   */
+  async _expandWildcardRepoRefs(repoRefs: git.GitHubRepoRef[]):
+      Promise<git.GitHubRepoRef[]> {
+    const ownersToFetchRepoNamesFor: Set<string> = new Set();
+    for (const repo of repoRefs) {
+      if (repo.repoName.match(/\*/)) {
+        ownersToFetchRepoNamesFor.add(repo.ownerName.toLowerCase());
+      }
+    }
+    if (ownersToFetchRepoNamesFor.size === 0) {
+      return Array.from(repoRefs);
+    }
+
+    // TODO(usergenic): When there are repos and tests with wildcards, we
+    // get two progress bars, identically labeled.  We should move the work to
+    // fetch the pages of repos into a support method that can be called in
+    // advance of the expand call and put the progress bar message there.
+    const allGitHubRepoRefs: git.GitHubRepoRef[] =
+        (await util.promiseAllWithProgress(
+             Array.from(ownersToFetchRepoNamesFor)
+                 .map(owner => this._github.getRepoFullNames(owner)),
+             'Fetching repo names for wildcard search'))
+            .reduce((a, b) => a.concat(b))
+            .map(git.parseGitHubRepoRefString);
+    const expandedRepoRefs: git.GitHubRepoRef[] = [];
+    for (const repoRef of repoRefs) {
+      if (repoRef.repoName.match(/\*/)) {
+        expandedRepoRefs.push.apply(
+            expandedRepoRefs,
+            allGitHubRepoRefs.filter(
+                otherRepoRef => git.matchRepoRef(repoRef, otherRepoRef)));
+      } else {
+        expandedRepoRefs.push(repoRef);
+      }
+    }
+    return expandedRepoRefs;
+  }
+
+  /**
+   * Cleans up the workspace folder and fixes repos which may be in
+   * incomplete
+   * or bad state due to prior abended runs.
+   */
+  async _prepareWorkspaceFolder() {
+    const workspaceDir = this._workspace.dir;
+
+    if (this._verbose) {
+      console.log(`Preparing workspace folder ${workspaceDir}...`);
+    }
+    // Clean up repos when 'fresh' option is true.
+    if (this._fresh) {
+      if (this._verbose) {
+        console.log(`Removing workspace folder ${workspaceDir}...`);
+      }
+      await promisify(rimraf)(workspaceDir);
+    }
+
+    // Ensure repos folder exists.
+    if (!util.existsSync(workspaceDir)) {
+      if (this._verbose) {
+        console.log(`Creating workspace folder ${workspaceDir}...`);
+      }
+      fs.mkdirSync(workspaceDir);
+    }
+
+    // Sometimes a repo will be left in a bad state. Deleting it here
+    // will let it get cleaned up later.
+    for (let dir of fs.readdirSync(workspaceDir)) {
```

Comment:
Is this using some heuristic to determine a bad repo? I can't quite parse what's going on.